app/helpers/application_helper.rb

```diff
@@ -1,4 +1,6 @@
 module ApplicationHelper
+  DELETED_USER = User.new(username: "[deleted user]", name: "[Deleted User]")
```

Comment:
Instead of creating an actual user instance, we could also create an `OpenStruct`. This could have the extra advantage that we can also override things that are defined as methods for `User`, so we don't need the conditional check for `path` in the Liquid tag:

```ruby
OpenStruct.new(username: "[deleted user]", name: "[Deleted User]", path: nil)
```

Pros:
* Removes conditional logic
* Addresses @vaidehijoshi's concerns about using an actual `User` object.

Con:
* Needs to be kept up to date with how we use this null object, though to a certain degree that's always true.

Reply:
There's definitely a part of me that likes this idea and considered it, but another part of me wasn't sure it was desirable to create an `OpenStruct` with as many keys as there are methods called on the user (`:username`, `:name`, `:path`, `:summary`, `:twitter_username`, `:github_username`). Still, I don't love having the unpersisted `User` object either so perhaps this makes sense.

Reply:
I don't think we need keys for all methods of `User`. If we have a  generic `NullObject` to inherit from that sets a couple of sensible defaults for things like `persisted?` I'd be happy to only have things we actually used defined on the specific objects.